src/chocolatey.resources/helpers/functions/Get-WebFile.ps1

```diff
@@ -177,17 +177,17 @@ param(
     $req.UserAgent = $userAgent
   }
 
-  if ($options.Headers.Count -gt 0) {
+  if($options.Headers.Count -gt 0){
     Write-Debug "Setting custom headers"
-    foreach ($item in $options.Headers.GetEnumerator()) {
-      $uri = (new-object system.uri $url)
-      Write-Debug($item.Key + ':' + $item.Value)
-      switch ($item.Key) {
-        'Accept' {$req.Accept = $item.Value}
-        'Cookie' {$req.CookieContainer.SetCookies($uri, $item.Value)}
-        'Referer' {$req.Referer = $item.Value}
-        'User-Agent' {$req.UserAgent = $item.Value}
-        Default {$req.Headers.Add($item.Key, $item.Value)}
+    foreach($key in $options.headers.keys){
+      $uri = (New-Object sysetem.uri $url)

```

Comment:
```suggestion
      $uri = New-Object -TypeName System.Uri $url

```

Reply:
If we are being pedantic I'll just use:

`$uri = [System.Uri]$url`

;)

Reply:
Fixed.